nixos/modules/services/development/fossil-server.nix

```diff
@@ -0,0 +1,146 @@
+          repositories with names ending in ".fossil" or a single file.
+        '';
+    };
+
+    baseurl = mkOption {
+      type = types.str;
+      default = "";
+      example = "example.com";
+      description = "Use URL as the base (useful for reverse proxies)";
+    };
+
+    extroot = mkOption {
+      type = types.str;
+      default = "";
+      description = "Document root for the /ext extension mechanism";
+    };
+
+    files = mkOption {
+      type = types.str;
+      default = "";
+      description = "Comma-separated list of glob patterns for static files";
+    };
+
+    https = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+          Indicates that the input is coming through a reverse proxy
+          that has already translated HTTPS into HTTP.
+        '';
+    };
+
+    jsmode = mkOption {
+      type = types.str;
+      default = "";
+      example = "inline";
+      description = ''
+          Determine how JavaScript is delivered with pages.
+          Mode can be one of:
+            inline       All JavaScript is inserted inline at
+                         the end of the HTML file.
+            separate     Separate HTTP requests are made for
+                         each JavaScript file.
+            bundled      One single separate HTTP fetches all
+                         JavaScript concatenated together.
+        '';
+    };
+
+    maxLatency = mkOption {
+      type = types.int;
+      default = 0;
+      example = 60;
+      description = "Do not let any single HTTP request run for more than N seconds";
+    };
+
+    nocompress = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Do not compress HTTP replies";
+    };
+
+    nojail = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Drop root privileges but do not enter the chroot jail";
+    };
+
+    nossl = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Signal that no SSL connections are available";
+    };
+
+    port = mkOption {
+      type = types.int;
+      default = 8080;
+      description = "Port to listen on";
+    };
+
+    repolist = mkOption {
+      type = types.bool;
+      default = if (lib.strings.hasSuffix ".fossil" cfg.repository) then false else true;
+      description = ''If REPOSITORY is dir, URL "/" lists repos.'';
+    };
+
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+
+    networking.firewall.allowedTCPPorts = [ cfg.port ];
+    systemd.services.fossil = {
+      preStart = if lib.strings.hasSuffix ".fossil" cfg.repository
+                 then ""
+                 else  ''
+                   if [ ! -d ${cfg.repository} ]; then
+                     mkdir -pv ${cfg.repository};
+                   fi
+                 '';
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      script = "${pkgs.fossil}/bin/fossil server "
```

Comment:
You can just use `ExecStart` instead.

Reply:
Yeah, it's nice how it just embeds it in the `.service` file instead of creating a script in the store. Thanks.